language/tools/move-mv-llvm-compiler/tests/rbpf-tests/entry-point06.json

```diff
@@ -0,0 +1,20 @@
+{
+    "program_id": "DozgQiYtGbdyniV2T74xMdmjZJvYDzoRFFqw7UR5MwPK",
+    "accounts": [
+        {
+            "key": "2ZE8gNT4wSpqvA2mUeYp7qpywQzFnmKHXgUrKyyVDGkt",
+            "owner": "BPFLoaderUpgradeab1e11111111111111111111111",
+            "is_signer": false,
+            "is_writable": true,
+            "lamports": 0,
+            "data": []
+        }
+    ],
+    "instruction_data": [
+        23, 0, 0, 0, 0, 0, 0, 0,
+        101, 110, 116, 114, 121, 95, 112, 111, 105, 110, 116, 95, 95, 109, 105, 115, 97, 108, 105, 103, 110, 101, 100,
+        1,
+        2, 3, 5, 7, 11, 13, 17, 19,
```

Comment:
One question I have is how does the second argument load (y) have proper alignment since these are packed? I looked at the generated code, and it seems possible that addresses will not necessarily be type-aligned. Does the VM automatically support misaligned accesses, or should we see a runtime exception if rbpf tries to load unaligned? Or, are we getting lucky here with the returned data?

Reply:
I think in this particular case the int64 is being properly aligned actually, because the string with the function name is 23 bytes and the boolean after that adds another byte. I changed the function name slightly (added another character) and now the address of parameter y is not aligned on the 8-byte boundary. This appears to work ok. The VM must be working around misaligned data accesses.